### PR TITLE
rapids to 23.06

### PR DESCRIPTION
**Pull request by eordentlich**
local (fast) tests pass, need to verify cloud.
Note that dataproc image had to be bumped to get python to > 3.8 which is dropped in rapids 23.06.
rapids-4-spark plugin 23.06 update pending push to maven.

### CODE REVIEW

**Comment by lijinf2** [comment]
Looks good overall. 

Does dataproc work with RAPIDS-23.06? Only the RAPIDS version gets upgraded, and python version remains the same (python 3.8?).

File: docker/Dockerfile [hunk 1]
```diff
@@ -88,7 +88,7 @@ RUN wget --quiet \
 # install cuDF dependency, Fall back to use cudf 22.04 due to issue:
 # https://github.com/NVIDIA/spark-rapids-ml/issues/73
 ARG CONDA_CUDF_VER=22.04
```
Comment (wbo4958) [inline docker/Dockerfile:90]: cudf is still 22.04?

**Comment by wbo4958** [comment]
I'm wondering how you make this PR.  seems there're lots of places need to be updated.

File: python/benchmark/databricks/init-pip-cuda-11.8.sh [hunk 2]
```diff
@@ -2,10 +2,10 @@
...
 BENCHMARK_ZIP=/dbfs/path/to/benchmark.zip
-# IMPORTANT: specify rapids fully 23.4.0 and not 23.4
-# also RAPIDS_VERSION (python) fields should omit any leading 0 in month/minor field (i.e. 23.4.0 and not 23.04.0)
-# while SPARK_RAPIDS_VERSION (jar) should have leading 0 in month/minor (e.g. 23.04.0 and not 23.4.0)
-RAPIDS_VERSION=23.4.0
+# IMPORTANT: specify rapids fully 23.6.0 and not 23.6
+# also RAPIDS_VERSION (python) fields should omit any leading 0 in month/minor field (i.e. 23.6.0 and not 23.06.0)
+# while SPARK_RAPIDS_VERSION (jar) should have leading 0 in month/minor (e.g. 23.06.0 and not 23.6.0)
+RAPIDS_VERSION=23.6.0
 SPARK_RAPIDS_VERSION=23.04.0
```
Comment (leewyang) [inline python/benchmark/databricks/init-pip-cuda-11.8.sh]: Should SPARK_RAPIDS_VERSION be updated as well?

Comment (eordentlich) [reply python/benchmark/databricks/init-pip-cuda-11.8.sh]: Yes it should.   My diff shows it is updated though.

**Comment by eordentlich** [comment]
build

**Comment by eordentlich** [comment]
Verified that after patch-release updates of rapids pip packages, benchmarks run on databricks, dataproc and emr.
Dataproc required patching pyarrow in the conda environment, update to 2.1, and a minor edit to the spark-rapids install script which was added to our dataproc benchmark script directory.

**Comment by eordentlich** [comment]
build

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
re-triggered, seems some internal networking issue
```
The TLS connection was non-properly terminated
```

**Comment by eordentlich** [comment]
build

**Comment by pxLi** [comment]
failed 
```
[2023-07-10T04:43:03.414Z] [0m[91mTraceback (most recent call last):
[2023-07-10T04:43:03.414Z]   File "/root/miniconda3/bin/mamba", line 7, in <module>
[2023-07-10T04:43:03.414Z]     from mamba.mamba import main
[2023-07-10T04:43:03.414Z]   File "/root/miniconda3/lib/python3.8/site-packages/mamba/mamba.py", line 53, in <module>
[2023-07-10T04:43:03.414Z]     import mamba.mamba_api as api
[2023-07-10T04:43:03.414Z] ImportError: libarchive.so.13: cannot open shared object file: No such file or directory
[2023-07-10T04:43:03.414Z] [0mThe command '/bin/sh -c conda install -c conda-forge mamba &&     mamba install -y -c rapidsai -c nvidia -c conda-forge cuml=$CUML_VER python=3.9 cuda-toolkit=11.5     && mamba clean --all -f -y' returned a non-zero code: 1
```

we saw this often in nightly docker images build, seems some bad timing about mamba cache layer, could try later

**Comment by pxLi** [comment]
build

**Comment by eordentlich** [comment]
build

**Comment by pxLi** [comment]
build

**Comment by eordentlich** [comment]
build

**Comment by lijinf2** [comment]
It seems have to copy third-party code in spark_rapids.sh. Will be wonderful if there is a way to avoid copy (can be addressed in the future). Overall looks good.

**Review by lijinf2 — APPROVED** [review]
Thanks for testing all of databricks, gcp dataproc and aws emr. Looks nice!

**Pull request merged**